packages/unmock-core/src/__tests__/nock.test.ts

```diff
@@ -0,0 +1,435 @@
+        type: "string",
+        enum: ["bar"],
+      });
+    });
+    it("A query in the path correctly propagated", () => {
+      expectNServices(0);
+      unmock
+        .nock("https://www.foo.com", "foo")
+        .get("/?q&m=1&") // include an empty query
+        .reply(200);
+      expectNServices(1);
+      expect(getPrivateSchema("foo").paths["/"].parameters[0].in).toEqual(
+        "query",
+      );
+      expect(getPrivateSchema("foo").paths["/"].parameters[0].name).toEqual("");
+      expect(getPrivateSchema("foo").paths["/"].parameters[0].schema).toEqual({
+        type: "null",
+      });
+      expect(getPrivateSchema("foo").paths["/"].parameters[1].in).toEqual(
+        "query",
+      );
+      expect(getPrivateSchema("foo").paths["/"].parameters[1].name).toEqual(
+        "m",
+      );
+      expect(getPrivateSchema("foo").paths["/"].parameters[1].schema).toEqual({
+        type: "string",
+        enum: ["1"],
+      });
+    });
+  });
+
+  describe("Request headers can be specified", () => {
+    it("An empty request header results in a viable spec", () => {
+      expectNServices(0);
+      unmock
+        .nock("https://www.foo.com", {}, "foo")
+        .get("/")
+        .reply(200);
+      expectNServices(1);
+      expect(Object.keys(getPrivateSchema("foo").paths)).toEqual(["/"]);
+    });
+
+    it("A request header is correctly propagated", () => {
+      expectNServices(0);
+      unmock
+        .nock("https://www.foo.com", { reqheaders: { hello: "world" } }, "foo")
+        .get("/")
+        .reply(200);
+      expectNServices(1);
+      expect(getPrivateSchema("foo").paths["/"].parameters[0].in).toEqual(
+        "header",
+      );
+      expect(getPrivateSchema("foo").paths["/"].parameters[0].name).toEqual(
+        "hello",
+      );
+      expect(getPrivateSchema("foo").paths["/"].parameters[0].schema).toEqual({
+        type: "string",
+        enum: ["world"],
+      });
+    });
+  });
+
+  describe("tldr adds 9 levels of arbitrary paths to requests", () => {
+    it("ignores unimportant stuff", () => {
+      expectNServices(0);
+      unmock.nock("https://www.foo.com", "foo").tldr();
+      expectNServices(1);
+      expect(Object.keys(getPrivateSchema("foo").paths).length).toEqual(9);
+    });
+  });
+
+  describe("Reply headers can be specified", () => {
+    it("An empty reply header results in a viable spec", () => {
+      expectNServices(0);
+      unmock
+        .nock("https://www.foo.com", "foo")
+        .get("/")
+        .reply(200, "", {});
+      expectNServices(1);
+      expect(Object.keys(getPrivateSchema("foo").paths)).toEqual(["/"]);
+    });
+
+    it("A reply header is correctly propagated", () => {
+      expectNServices(0);
+      unmock
+        .nock("https://www.foo.com", "foo")
+        .get("/")
+        .reply(200, "", { hello: "world" });
+      expectNServices(1);
+      expect(
+        getPrivateSchema("foo").paths["/"].get.responses["200"].headers.hello
```

Comment:
I'm wondering if we can use more meaningful placeholders than "foo", "bar", etc. 🤔 That way, the tests better explain the functionality and maybe even enable onboarding into the project.

This doesn't need to be part of this PR as you were moving these tests, not writing them from scratch - but something to consider.

Reply:
Definitely!